### PR TITLE
Add additional unit tests of the `/locations` route, focusing on testing mocks

### PR DESCRIPTION
**Pull request by dvwhite**
I added additional assertions to the existing unit tests of the `/locations` controllers that utilized mocks to avoid interacting with the database in the tests

* Added an assertion to check that the mocks were called once and only once in each test that utilized `jest.mock.mockResolvedValue`

For instance, this assertion is used to test a model function called `findAll` that we mocked in the location controller tests:
```
// Verify that the mock was called
expect(Locations.findAll.mock.calls.length).toBe(1);
```
I also verify the arguments passed to the mocked functions in additional test assertions.

I also created a test object factory module called `generate.js` that contains a collection of functions that generate test objects or data using the `faker` library. The test objects are structured as per our test, model and migrations for the `/locations` route. It is easy to extend and use it for other libraries. 

Using the generated test objects should help with the maintainability of the tests and reduce the time spent hardcoding test objects.

I also added a unit test of the route handlers for POST requests send to `/locations`. While testing the route, I uncovered a bug with the route handler in which it was looking for a groomer id in the route params when no groomer id was used in the route params. I grabbed it from the request object instead.

While testing this route, I also uncovered a bug in our migration for the locations table: it was set to autoincrement. I explored several options to fix this in a migration rather than editing the original migration file. Unfortunately, the only reliable fixes involved either adding and using a sequence, and then manually incrementing the record id on every insertion within the model by calling `NEXTVAL('sequence_name')`, or, simply grabbing the last record id used and incrementing it by one in the model. 

It was not possible to add autoincrementing to the id column after creating it, nor was it possible to drop the column and add it back with the autoincrementing set on the table field. I would have preferred to add another migration to edit this rather than modifying the original migration that created the table, however I thought it was actually worse to use the workarounds and then have an inconsistent API for the app's router models.

### CODE REVIEW

**Review by Cberumen51 — DISMISSED** [review]
Good use on the in line comments to help identify what the test does

**Review by Cberumen51 — COMMENTED** [review]
The utils look great the changes clean up the testing real nice

File: __tests__/routes/locations.test.js [hunk 1]
```diff
@@ -25,79 +35,120 @@ describe('locations router endpoints', () => {
...
-          email: 'Carolanne.Ullrich14@gmail.com',
-          phoneNumber: '414-962-7162',
-          lat: -53.91525,
-          lng: -96.60353,
-        },
-      ]);
-      const res = await request(server).get('/locations/bftqzfqkq3xzj953q64r');
+      const location = generate.buildLocation();
+      Locations.findBy.mockResolvedValue([location]);
+      const res = await request(server).get(`/locations/${location.groomerId}`);
```
Comment (hmerritt) [inline __tests__/routes/locations.test.js:48]: Mocking using the object factory functions look **_much_** cleaner!

File: test/utils/generate.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    businessName: getLocationName(),
+    address: getAddress(),
+    city: getCity(),
+    state: getState(),
+    zip: getZipCode(),
+    email: getEmail(),
+    phoneNumber: getPhoneNumber(),
+    lat: getLat(),
+    lng: getLng(),
+    ...overrides,
```
Comment (hmerritt) [inline test/utils/generate.js:53]: Brilliant idea to generate random mock data

**Review by hmerritt — APPROVED** [review]
Great job on generating random mock data for the tests and adding migration data for locations table.

File: __tests__/routes/locations.test.js (hunk 1 above)
Comment (Cberumen51) [reply __tests__/routes/locations.test.js:48]: Way to condense the code so it doesn't take look so bloated

**Review by Cberumen51 — APPROVED** [review]
The code looks a lot cleaner and the util look great

**Pull request merged**